dev/wait.go

```diff
@@ -0,0 +1,213 @@
+package dev
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+)
+
+const (
+	WaiterDefaultTimeout  = 60 * time.Second
```

Comment:
Out of curiosity, why 1 minute? Seems like an eternity lol

Reply:
When checking for Readiness images need to pulled from the internet, 30s was definitely too low so I doubled it.
It very much depends on what the user is doing, so I think most users will need to override this setting anyway.
If you need to override, you can override globally and also for every `Wait*` call, just noticed I should expose that for the `CreateAndWait` methods as well.